aws_utils/s3/s3_utils.py

```diff
@@ -410,3 +411,53 @@ def get_contents_of_directory(directory, bucket=None):
     bucket = get_bucket(bucket)
 
     return [x.key for x in bucket.list(prefix=directory)]
+
+
+def rename_s3_key(boto3_client, bucket_name, old_prefix, new_prefix):
+    """
+    Rename a single file on S3 (=> copy + delete)
+    Args:
+        boto3_client (botocore.client.S3) : boto3 client
+        bucket_name (str):
+        old_prefix (str):
+        new_prefix (str):
+    """
+    if old_prefix != new_prefix:
+        if not bucket_name:
+            raise (ClientError, AttributeError)
+        boto3_client.copy_object(CopySource={'Bucket': bucket_name, 'Key': old_prefix}, Bucket=bucket_name, Key=new_prefix)
+        boto3_client.delete_object(Bucket=bucket_name, Key=old_prefix)
+        logger.info('Moved key {}'.format(old_prefix))
+    else:
+        logger.warn('Source and destination paths are the same')
+
+
+def rename_keys_on_s3(bucket_name, bucket_region, prefix_root, prefix_modification_func,
+                      filter_keys_func=None):
+    """
+    Renames all keys in a prefix_root/folder/ which are not filtered by a filter function
+    Args:
+        bucket_name (str): An existing s3 bucket
+        bucket_region (str): A valid s3 region e.g 'us-east-1'
+        prefix_root (str): The prefix to start the recursive renaming with
+        prefix_modification_func (function): A function that changes the prefix string. In case it is None, nothing will be moved
+        filter_keys_func (function): A function that will apply a filter to the keys we don't want to move. In case it is None, nothing will be moved
+    """
```

Comment:
Would be really good to give an example here, use the googledocs format for the example is

Example: